netmaster/objApi/apiController.go

```diff
@@ -0,0 +1,639 @@
+/***
+Copyright 2014 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package objApi
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/contiv/netplugin/netmaster/intent"
+	"github.com/contiv/netplugin/netmaster/master"
+	"github.com/contiv/netplugin/utils"
+	"github.com/contiv/objmodel/contivModel"
+	"github.com/contiv/objmodel/objdb/modeldb"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gorilla/mux"
+)
+
+// APIController stores the api controller state
+type APIController struct {
+	router *mux.Router
+}
+
+var apiCtrler *APIController
+
+// NewAPIController creates a new controller
+func NewAPIController(router *mux.Router) *APIController {
+	ctrler := new(APIController)
+	ctrler.router = router
+
+	// initialize the model objects
+	contivModel.Init(ctrler)
+
+	// Register routes
+	contivModel.AddRoutes(router)
+
+	// Add default tenant if it doesnt exist
+	tenant := contivModel.FindTenant("default")
+	if tenant == nil {
+		log.Infof("Creating default tenant")
+		err := contivModel.CreateTenant(&contivModel.Tenant{
+			Key:        "default",
+			TenantName: "default",
+			SubnetPool: "10.1.1.1/16",
+			SubnetLen:  24,
+			Vlans:      "1-2000",
+			Vxlans:     "10001-12000",
+		})
+		if err != nil {
+			log.Fatalf("Error creating default tenant. Err: %v", err)
+		}
+	}
+
+	return ctrler
+}
+
+// AppCreate creates app state
+func (ac *APIController) AppCreate(app *contivModel.App) error {
```

Comment:
Ignore all objects except tenant and network for now. I'll remove all the objects that are not needed in net master in a subsequent commit.